cfme/utils/template/ec2.py

```diff
@@ -49,25 +49,47 @@ def upload_image(self):
 
     @log_wrap("import image from bucket")
     def import_image(self):
-        try:
-            import_task_id = self.mgmt.import_image(s3bucket=self.bucket_name,
-                                                    s3key=self.template_name,
-                                                    description=self.template_name)
-
-            wait_for(self.mgmt.get_image_id_if_import_completed,
-                     func_args=[import_task_id],
-                     fail_condition=False,
-                     delay=5,
-                     timeout='90m',
-                     message='Importing image to EC2')
-
-            ami_id = self.mgmt.get_image_id_if_import_completed(import_task_id)
-            ami = EC2Image(uuid=ami_id, system=self.mgmt)
-            ami.set_tag("Name", self.template_name)
-            return True
+        if self.stream == 'downstream-510z':
+            try:
+                import_task_id = self.mgmt.import_image(s3bucket=self.bucket_name,
+                                                        s3key=self.template_name,
+                                                        description=self.template_name)
+
+                wait_for(self.mgmt.get_image_id_if_import_completed,
+                         func_args=[import_task_id],
+                         fail_condition=False,
+                         delay=5,
+                         timeout='90m',
+                         message='Importing image to EC2')
+
+                ami_id = self.mgmt.get_image_id_if_import_completed(import_task_id)
+                ami = EC2Image(uuid=ami_id, system=self.mgmt)
+                ami.set_tag("Name", self.template_name)
+                return True
 
-        except Exception:
-            return False
+            except Exception:
+                return False
+        else:
+            try:
+                import_task_id = self.mgmt.import_snapshot(s3bucket=self.bucket_name,
+                                                        s3key=self.template_name,
+                                                        description=self.template_name)
+                wait_for(self.mgmt.get_snapshot_id_if_import_completed,
+                         func_args=[import_task_id],
+                         fail_condition=False,
+                         delay=5,
+                         timeout='90m',
+                         message='Importing image to EC2')
+
+                snapshot_id = self.mgmt.get_snapshot_id_if_import_completed(import_task_id)
+                ami_result = self.mgmt.create_image_from_snapshot(self.template_name, snapshot_id)
+                ami_id = ami_result.get("ImageId")
+                ami = EC2Image(uuid=ami_id, system=self.mgmt)
```

Comment:
From here down they seem to also be the same. I think you can de-dup this a bit.